### PR TITLE
fix: remove granular status feature flag

### LINKED ISSUE #33287: [Task]: Remove granular status feature flag
### SubTasks

- Remove `release_git_status_granular_enabled`

### PR DESCRIPTION
**Pull request by brayn003**
## Description
Removes `release_git_status_granular_enabled` feature flag

Fixes #33287 

## Automation

/ok-to-test tags="@tag.Git"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9018396111>
> Commit: a2af6687222992a2b5884a4c6086ad95b951535c
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9018396111&attempt=1" target="_blank">Click here!</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

### CODE REVIEW

File: app/client/src/pages/Editor/gitSync/components/GitChangesList/index.tsx [hunk 1]
```diff
@@ -1,16 +1,63 @@
...
+const CalloutContainer = styled.div`
+  margin-top: 16px;
+`;
+
+const DummyChange = styled.div`
+  width: 50%;
+  height: 18px;
+  background: linear-gradient(
+    90deg,
+    ${Colors.GREY_2} 0%,
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/gitSync/components/GitChangesList/index.tsx:23]: can we use a CSS variable from ADS instead?

Also, why is the component named `DummyChange`?

Comment (brayn003) [reply app/client/src/pages/Editor/gitSync/components/GitChangesList/index.tsx]: Done

**Review by ankitakinger — DISMISSED** [review]

File: app/client/src/pages/Editor/gitSync/components/GitChangesList/index.tsx [hunk 2]
```diff
@@ -1,16 +1,62 @@
...
+    var(--ads-v2-color-black-75) 0%,
+    rgba(240, 240, 240, 0) 100%
   );
-  if (isGitGranularFeatureEnabled) {
-    return <NewChangesList />;
-  } else {
-    return <OldChangesList />;
+  margin-top: 16px;
+  margin-bottom: 16px;
+`;
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/src/pages/Editor/gitSync/components/GitChangesList/index.tsx:27]: This is not required!

File: app/client/src/pages/Editor/gitSync/components/GitChangesList/index.tsx [hunk 3]
```diff
@@ -2,15 +1,62 @@
...
+      <PageChanges status={status} />
+      <ExpandableChange
+        kind={ExpandableChangeKind.DATASOURCES}
+        status={status}
+      />
+      <ExpandableChange kind={ExpandableChangeKind.JSLIBS} status={status} />
+      <StaticChage kind={StaticChangeKind.SETTINGS} status={status} />
+      <StaticChage kind={StaticChangeKind.THEME} status={status} />
+      <StaticChage kind={StaticChangeKind.PACKAGES} status={status} />
+      <StaticChage kind={StaticChangeKind.MODULES} status={status} />
```
[the remark this answers is not in the document]
Comment (brayn003) [reply app/client/src/pages/Editor/gitSync/components/GitChangesList/index.tsx]: this is resolved

**Review by ankitakinger — APPROVED** [review]

**Pull request merged**